packages/backend/src/rates/service.ts

```diff
@@ -51,22 +53,42 @@ class RatesServiceImpl implements RatesService {
     this.cachedRates = createInMemoryDataStore(deps.exchangeRatesLifetime)
   }
 
-  async convert(
-    opts: Omit<ConvertOptions, 'exchangeRate'>
-  ): Promise<bigint | ConvertError> {
-    const sameCode = opts.sourceAsset.code === opts.destinationAsset.code
-    const sameScale = opts.sourceAsset.scale === opts.destinationAsset.scale
-    if (sameCode && sameScale) return opts.sourceAmount
-    if (sameCode) return convert({ exchangeRate: 1.0, ...opts })
+  async convert(opts: RateConvertOpts): Promise<ConvertResults | ConvertError> {
+    const {
+      reverseDirection = false,
```

Comment:
convert function is same as before except this `reverseDirection` controls which conversion function is used (`convert` or new `convertReverse`). Alternatively considered making a different `ratesService.convertReverse` - perhaps thats a bit more explicit but I didn't have a strong preference and found this to be simpler.